src/main/java/reactor/core/publisher/BlockingSink.java

```diff
@@ -278,14 +280,11 @@ public Emission finish() {
 		}
 	}
 
-	@Override
 	public long getCapacity() {
-		return Trackable.class.isAssignableFrom(actual.getClass()) ? ((Trackable)
-				actual).getCapacity() :
-				Long.MAX_VALUE;
+		return Scannable.from(actual)
+		                .scanOrDefault(Scannable.Attr.CAPACITY, Integer.MAX_VALUE);
```

Comment:
it does not close the issue which is related to Flux#getPrefetch (public). But no capacity currently will return Integer.MAX_VALUE. In fact BlockingSink is deprecated itself.